### PR TITLE
SONARJAVA-1142 Fix issue with packages ending in java/lang

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

**Comment by benzonico** [comment]
@m-g-sonar please note I have also factorized a bit of pacakge utils.

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java [hunk 1]
```diff
@@ -130,13 +128,15 @@ public void visitFile(@Nullable Tree parsedTree) {
...
-    boolean isJavaLangAnnotation = path.length > 4 && "Annotation.java".equals(path[path.length - 1]) && "java".equals(path[path.length - 4])
-      && "lang".equals(path[path.length - 3]) && "annotation".equals(path[path.length - 2]);
-    boolean isSerializable = path.length > 3 && "Serializable.java".equals(path[path.length - 1]) && "java".equals(path[path.length - 3]) && "io".equals(path[path.length - 2]);
-    return !(isJavaLang || isJavaLangAnnotation || isSerializable);
+  private boolean isNotJavaLangOrSerializable(String packageName) {
+    if(packageName.equals("java/lang")) {
+      return false;
+    } else if(packageName.equals("java/lang/annotation") && getContext().getFile().getName().equals("Annotation.java")) {
+      return false;
+    } else if(packageName.equals("java/io") && getContext().getFile().getName().equals("Serializable.java")) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: MAJOR Move the "Serializable.java" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
MAJOR Move the "java/io" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java [hunk 2]
```diff
@@ -130,13 +128,15 @@ public void visitFile(@Nullable Tree parsedTree) {
...
 
-  private boolean isNotJavaLangOrSerializable() {
-    String[] path = getContext().peekSourceCode().getName().split(Pattern.quote(File.separator));
-    boolean isJavaLang = path.length > 3 && "java".equals(path[path.length - 3]) && "lang".equals(path[path.length - 2]);
-    boolean isJavaLangAnnotation = path.length > 4 && "Annotation.java".equals(path[path.length - 1]) && "java".equals(path[path.length - 4])
-      && "lang".equals(path[path.length - 3]) && "annotation".equals(path[path.length - 2]);
-    boolean isSerializable = path.length > 3 && "Serializable.java".equals(path[path.length - 1]) && "java".equals(path[path.length - 3]) && "io".equals(path[path.length - 2]);
-    return !(isJavaLang || isJavaLangAnnotation || isSerializable);
+  private boolean isNotJavaLangOrSerializable(String packageName) {
+    if(packageName.equals("java/lang")) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: MAJOR Move the "java/lang" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java [hunk 3]
```diff
@@ -130,13 +128,15 @@ public void visitFile(@Nullable Tree parsedTree) {
...
+      return false;
+    } else if(packageName.equals("java/lang/annotation") && getContext().getFile().getName().equals("Annotation.java")) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: MAJOR Move the "java/lang/annotation" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)
MAJOR Move the "Annotation.java" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.

**Comment by m-g-sonar** [comment]
LGTM!

**Pull request merged**